Add OAuth contoller and GitHubService.

dcharoulis opened the pull request:
Improved comments in project.

mahyaa committed:
adds compile time dependency injection

Commit:
Add OAuth contoller and GitHubService.
Improved comments in project.

erikwj commented on app/controllers/OAuthGitHubController.scala:
```diff
@@ -0,0 +1,66 @@
+    */
+  def success() = Action.async { request =>
+    request.session.get("oauth-token").fold(Future.successful(Unauthorized("No way Jose"))) { authToken =>
+      cacheApi.set("authToken", authToken)
```
When does the cache expire?

erikwj commented on app/services/GitHubService.scala:
```diff
@@ -0,0 +1,174 @@
+                   branches_url:String,
+                   tags_url:String,
+                   blobs_url:String,
+                   //git_tags_url:String,
```
can these commented-out fields be deleted?

dcharoulis replied:
Sure, i am going to remove the commented ones. Actually maybe we can remove more in the future if they are not going to be used.

mariadroman committed:
Merge pull request #19 from mahyaa/master

adding cache to controller

Commit:
Fix issues
- remove unucessary comments from code
- use session instread of cache for "oauth-token"

mahyaa commented on app/controllers/OAuthGitHubController.scala:
```diff
@@ -0,0 +1,64 @@
+      )
+    }
+  }
+}
```
Why there is no documentation for what does the method returns? (for all of the methods in this class)
